src/app/errorReports/newErrorReportForm/NewErrorReportForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { postErrorReport, getLocation } from '../../../actions';
+import { selectIsPostingErrorReport } from '../../../reducers';
+import { selectLocationError, selectLocationData } from '../../../selectors/location';
+import LoadingLabel from '../../../components/form/LoadingLabel';
+import ErrorLabel from '../../../components/form/ErrorLabel';
+import ErrorReportText from './ErrorReportText';
+import ErrorReportServices from './ErrorReportServices';
+
+class NewErrorReportForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errorReportText: '',
+      errorReportGeneralLocationError: false,
+      errorReportServices: [],
+      isErrorReportFinished: false,
+    };
+
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportFinished = this.onErrorReportFinished.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation(this.props.match.params.locationId);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isPostingErrorReport && !this.props.isPostingErrorReport) {
+      this.props.history.push(`/errorreports/${this.props.match.params.locationId}/thanks`);
+    }
+  }
+
+  onErrorReportTextChanged(text) {
+    this.setState({ errorReportText: text });
+  }
+
+  onErrorReportServicesChanged(value) {
+    this.setState((state) => {
+      const errorReportServices = state.errorReportServices.includes(value)
+        ? state.errorReportServices.filter(service => service !== value)
+        : [...state.errorReportServices, value];
+      return {
+        errorReportServices,
+      };
+    });
+  }
+
+  onErrorReportGeneralLocationErrorChanged(value) {
+    this.setState({ errorReportGeneralLocationError: value });
+  }
+
+  onErrorReportFinished() {
+    this.setState({ isErrorReportFinished: true });
+  }
+
+  onErrorReportSubmitted() {
+    console.log('onErrorReportSubmitted triggered.');
+
+    const { errorReportText, errorReportGeneralLocationError, errorReportServices } = this.state;
+
+    this.props.postErrorReport(
+      errorReportText,
+      errorReportGeneralLocationError,
+      errorReportServices,
+    );
+  }
+
+  render() {
+    const { isPostingErrorReport, locationError } = this.props;
+
+    if (locationError) {
+      return <ErrorLabel errorMessage={locationError} />;
+    }
+
+    // Comments component checks for (!locationData || isPostingErrorReport)
+    // Not sure what the !locationData check was doing specifically
```

Comment:
It's for while the location data is being fetched, to make sure a loading indicator is shown.
Try refreshing the page (preferably while using throttling in Chrome's dev tools), and see what happens while it's loading.

I think the only reason you're not seeing actual crashes without this check is that right now, location data is coming to the child components not from here, but rather from the `withErrorReportsForm`, which does have a `if (!locationData) {` check.